resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/features/BasicExecutorFeatureStep.scala

```diff
@@ -175,49 +138,41 @@ private[spark] class ExecutorPodFactory(
       .withPorts(requiredPorts.asJava)
       .addToArgs("executor")
       .build()
-
-    val executorPod = new PodBuilder()
-      .withNewMetadata()
-        .withName(name)
-        .withLabels(resolvedExecutorLabels.asJava)
-        .withAnnotations(executorAnnotations.asJava)
-        .withOwnerReferences()
-          .addNewOwnerReference()
-            .withController(true)
-            .withApiVersion(driverPod.getApiVersion)
-            .withKind(driverPod.getKind)
-            .withName(driverPod.getMetadata.getName)
-            .withUid(driverPod.getMetadata.getUid)
-            .endOwnerReference()
-        .endMetadata()
-      .withNewSpec()
-        .withHostname(hostname)
-        .withRestartPolicy("Never")
-        .withNodeSelector(nodeSelector.asJava)
-        .endSpec()
-      .build()
-
     val containerWithLimitCores = executorLimitCores.map { limitCores =>
       val executorCpuLimitQuantity = new QuantityBuilder(false)
         .withAmount(limitCores)
         .build()
       new ContainerBuilder(executorContainer)
         .editResources()
-        .addToLimits("cpu", executorCpuLimitQuantity)
-        .endResources()
+          .addToLimits("cpu", executorCpuLimitQuantity)
```

Comment:
Just curious. Do we not support executor CPU request?

Reply:
We do, it's set a few lines up.

Reply:
https://github.com/apache/spark/pull/20910/files/7d65875266ba94f27d2cc8ec992e8e1cb8f593b5#diff-309ffff414bee6f4741e43bd01e9a8fdR135

Reply:
I see it now. Thanks!